### PR TITLE
DOC-3697 Add ability to specify group access at unit level

### PR DESCRIPTION
**Pull request by catong**
## [DOC-3697](https://openedx.atlassian.net/browse/DOC-3697)

Update these topic areas to account for new ability to set group access at the unit level:
- developing course units
- developing course components
- differentiated content by enrollment track
- differentiated content by cohort

NB - includes reference to gear icon next to unit title on unit page, which is still in progress in https://openedx.atlassian.net/browse/EDUCATOR-542

### Reviewers
The PR submitter checks the boxes after each reviewer finishes and gives :+1:. 

- [x] Subject matter expert: @jlajoie 
- [x] Subject matter expert: @bbaker6225 
- [ ] Subject matter expert: @cahrens 
- [x] Doc team review: @edx/doc 
- [x] Product review: @sstack22 

FYI: @mmacfarlane @jhendersonedx, @dhixonedx

### HTML Version 
- [x] http://draft-ca-unit-level-group-access.readthedocs.io/en/latest/developing_course/course_units.html#set-access-restrictions-for-a-unit
- [x] http://draft-ca-unit-level-group-access.readthedocs.io/en/latest/course_features/diff_content/enroll_track_courseware.html#specify-content-as-available-only-to-particular-enrollment-tracks
- [x] http://draft-ca-unit-level-group-access.readthedocs.io/en/latest/course_features/cohorts/cohorted_courseware.html#specify-content-as-available-only-to-particular-content-groups

### Sandbox
- [ ] Point to or build a sandbox for the software change and add a link here

### Testing
- [x] Ran ./run_tests.sh without warnings or errors

### Post-review

- [ ] Add a comment with the description of this change or link this PR to the next release notes task.
- [ ] Squash commits

### CODE REVIEW

File: en_us/shared/developing_course/course_units.rst [hunk 1]
```diff
@@ -365,6 +379,80 @@ being moved to the top of the unit.
...
+    :alt: The visibility and access settings dialog for a unit, with a
+       dropdown list for selecting the type of learner group you will use for
+       restricting access.
+    :width: 200
+
+
+   .. note:: You can also access the **Restrict access to** option from the
+      **Access** icon next to a unit's name on a unit page in Studio.
+
+.. image to come
```
Comment (catong) [inline en_us/shared/developing_course/course_units.rst]: To come with https://openedx.atlassian.net/browse/EDUCATOR-542
https://github.com/edx/edx-platform/pull/15424

File: en_us/shared/course_features/cohorts/cohorted_courseware.rst [hunk 2]
```diff
@@ -150,114 +150,44 @@ You associate each content group with one or more cohorts in the LMS, on the
...
+to are available to all learners enrolled in your course, regardless of the
+cohort that they belong to.
 
-In a separate task, you create cohorts and content groups, and then associate
-each content group with one or more cohorts. Then, only the cohorts associated
-with content groups that you selected in a component's access settings
-can view the component. See :ref:`Associate Cohorts with Content Groups` for
-details about associating cohorts with content groups.
+.. note:: If a unit has group access restrictions set, all of its child
+   components inherit these group access restrictions unless you explicitly
```
Comment (bbaker6225) [inline en_us/shared/course_features/cohorts/cohorted_courseware.rst]: Do we want to add language clarifying that you can't override unit level access at the component access editor (just narrow it down)?  
For example, if you restrict a unit to Verified Learners and then set a component's access level to "All Learners and Staff", only verified learners will be able to see that component.
Same question for the next note on this as well

File: en_us/shared/developing_course/course_units.rst [hunk 3]
```diff
@@ -365,6 +379,80 @@ being moved to the top of the unit.
...
+***********************************
+Set Access Restrictions For a Unit
+***********************************
+
+If you have more than one enrollment track in your course, or if you have
+enabled cohorts, you can limit a unit's availability to specific groups of
+learners. For information about offering different content to different learner
+groups, see :ref:`Offering Differentiated Content`.
+
+.. note:: If a unit has group access restrictions set, all of its child
```
Comment (bbaker6225) [inline en_us/shared/developing_course/course_units.rst]: Do we want to add language to this note clarifying that component access settings can not override the unit level access settings?  
For example, if you restrict unit access to verified learners and then set a component's visibility to all learners and staff, the component will still only be visible to verified learners

**Review by bbaker6225 — COMMENTED** [review]
Everything looks good to me, I just had a question about whether we wanted to clarify the language about unit & component level access and how they interact.

File: en_us/shared/developing_course/course_units.rst (hunk 3 above)
Comment (catong) [reply en_us/shared/developing_course/course_units.rst]: @bbaker6225 Great point. Thanks for catching this! I'll rework the notes.

**Review by jlajoie — APPROVED** [review]

**Review by bbaker6225 — APPROVED** [review]

File: en_us/shared/developing_course/course_components.rst [hunk 4]
```diff
@@ -119,6 +120,76 @@ To set the display name for a component, follow these steps.
...
+   The groups which have access to the component are listed under the
+   component title in the Studio course outline.
+
+   .. image:: ../../../shared/images/component_access_indicator.png
+    :alt: When a component has restricted access, a message listing the groups
+      which have access to the component appears under the component title in
+      the Studio course outline.
+    :width: 500
+
+   In addition,on the unit page in Studio, a message in the publishing status bar
```
Comment (grantgoodmanedX) [inline en_us/shared/developing_course/course_components.rst]: add a space between addition, and on

**Review by grantgoodmanedX — APPROVED** [review]
This looks good to me, with one little nit. 👍

**Comment by sstack22** [comment]
👍 ! looks great, thanks!

**Comment by catong** [comment]
Merging https://github.com/edx/edx-documentation/pull/1528

**Pull request merged**